app/services/service_offering/add_to_portfolio_item.rb

```diff
@@ -27,25 +25,20 @@ def process
 
     private
 
-    def determine_valid_fields
-      # These are the fields that topology gave us, so we want to pull them into our model.
-      # Along the way we map them to their string counterparts, then reject the columsn we don't need
-      @service_offering.instance_variables
-                       .map { |var| var.to_s.delete("@") }
-                       .reject { |field| IGNORE_FIELDS.include?(field) }
-    end
-
     # Move fields from the ServiceOffering Ojbect to our params hash, from which we build our PortfolioItem
-    def create_param_map(fields)
-      # logical AND the column names with the fields we know we have, so we only pull in what is available
-      (PortfolioItem.column_names & fields).each do |column|
-        @params[column] = @service_offering.instance_variable_get("@" + column)
+    def create_param_map
+      PortfolioItem.column_names.each do |column|
+        next if IGNORE_FIELDS.include?(column)
+        if @service_offering.respond_to?(column)
```

Comment:
A few thoughts here:

1) Since `PortfolioItem.column_names` is a static list and the fields we are skipping is a constant, could they be combined into another constant?  Hopefully removing the need for the `next` line.

2) If we can get an array of property names coming from service_offerings we could use set logic to limit the results to just column names that already exist for PortfolioItems and skip the `respond_to?` check.

3) Should we create a new private method that can be used to determine the column names to iterate over which could be tested separately.